### DISCUSSION TITLE
Docker: Sys_Admin Cap Reason?

### DISCUSSION THREAD

**Discussion by jakobkukla** [jef/streetmerchant#988] [Help]
### Description

Hey,

Can I run the docker container without the SYS_ADMIN cap? I usually like to run my containers in unprivileged mode, since it otherwise defeats the whole purpose of sandboxing with containerization.

If not, I suspect there must be a specific reason for the privileged mode. Sorry if I'm missing something obvious. It's late around here...

Thanks

**Comment by perfectly-preserved-pie** [comment]
I'd like to know this too. I'm curious to know what part of the bot requires (almost) root access to the host.

**Comment by riskynacho** [comment]
I haven't ran this in docker but I suspect it for those mac users who want to receive external calls to play sounds and open browsers.

**Comment by jef** [comment]
I agree! I would really not like to have this run as privileged mode. Unfortunately, Chromium doesn't like running without these priveledges.

Take a look [here](https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md#tips).

> Seeing other weird errors when launching Chrome? Try running your container with docker run --cap-add=SYS_ADMIN when developing locally. Since the Dockerfile adds a pptr user as a non-privileged user, it may not have all the necessary privileges.

If you find anything in there and do some testing, I'd love to not have this requirement! Otherwise, it is a side effect with running in Docker.

**Reply by rchenzheng** [reply]
Thanks for your reply, I found this very concerning. Should we be looking at a solution to this as an issue?

**Reply by jef** [reply]
It's actually being worked on here now! https://github.com/jef/streetmerchant/pull/1291
